docs/index.md

```diff
@@ -9,22 +9,44 @@ description: |-
 
 The UpCloud Terraform Provider enables organisations to control resources provided by the UpCloud platform.
 
+## Configure provider
+### Using environment variables
+Define credentials using `UPCLOUD_USERNAME` and `UPCLOUD_PASSWORD` environment variables.  
+```terraform
+terraform {
+  required_providers {
+    upcloud = {
+      source  = "UpCloudLtd/upcloud"
+      version = "~> 2.0"
+    }
+  }
+}
+
+provider "upcloud" {
+  # username and password configuration arguments can be omitted  
+  # if environment variables UPCLOUD_USERNAME and UPCLOUD_PASSWORD are set
+  # username = ""
+  # password = ""
```

Comment:
I would also put this comment in the `minimal` example, as that is probably the first one users would check